src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java

```diff
@@ -0,0 +1,196 @@
+   */
+  @Parameter(property = "apiKey", required = true)
+  private String apiKey;
+
+  /**
+   * Service Key for communicating with Contrast. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal keys</a>
+   */
+  @Parameter(property = "serviceKey", required = true)
+  private String serviceKey;
+
+  /** Contrast API URL */
+  @Parameter(alias = "apiUrl", defaultValue = "https://app.contrastsecurity.com/Contrast")
+  private String url;
+
+  /**
+   * Unique ID for the Contrast Organization to which the plugin reports results. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your Organization ID</a>
+   */
+  // TODO[JG] must this be required? If a user is only in one org, we can look it up using the
+  // endpoint /ng/profile/organizations
+  @Parameter(alias = "orgUuid", required = true)
+  private String organizationId;
+
+  /**
+   * When true, will override Maven's proxy settings with Contrast Maven plugin specific proxy
+   * configuration
+   *
+   * @deprecated in a future release, we will remove the proprietary proxy configuration in favor of
+   *     standard Maven proxy configuration
+   */
+  @Parameter(property = "useProxy", defaultValue = "false")
+  private boolean useProxy;
+
+  /**
+   * Proxy host used to communicate to Contrast when {@code useProxy} is true
+   *
+   * @deprecated in a future release, we will remove the proprietary proxy configuration in favor of
+   *     standard Maven proxy configuration
+   */
+  @Parameter(property = "proxyHost")
+  private String proxyHost;
+
+  /**
+   * Proxy port used to communicate to Contrast when {@code useProxy} is true
+   *
+   * @deprecated in a future release, we will remove the proprietary proxy configuration in favor of
+   *     standard Maven proxy configuration
+   */
+  @Parameter(property = "proxyPort")
+  private int proxyPort;
+
+  String getUserName() {
+    return userName;
+  }
+
+  void setUserName(final String userName) {
+    this.userName = userName;
+  }
+
+  String getApiKey() {
+    return apiKey;
+  }
+
+  void setApiKey(final String apiKey) {
+    this.apiKey = apiKey;
+  }
+
+  String getServiceKey() {
+    return serviceKey;
+  }
+
+  void setServiceKey(final String serviceKey) {
+    this.serviceKey = serviceKey;
+  }
+
+  String getURL() {
+    return url;
+  }
+
+  void setURL(final String url) {
+    this.url = url;
+  }
+
+  String getOrganizationId() {
+    return organizationId;
+  }
+
+  void setOrganizationId(final String organizationId) {
+    this.organizationId = organizationId;
+  }
+
+  /**
+   * @return new ContrastSDK configured to connect with the authentication and proxy parameters
+   *     defined by this abstract mojo
+   * @throws MojoExecutionException when fails to connect to Contrast
+   */
+  ContrastSDK connectToContrast() throws MojoExecutionException {
+    Proxy proxy = getProxy();
+
+    try {
+      if (!StringUtils.isEmpty(url)) {
```

Comment:
Good catch. I added the default for the URL instead of delegating to the Contrast SDK to provide the default, because it felt more idiomatic and the default will then be apparent to users from the documentation. I'll remove this no longer needed logic